File: WandleWheelhouse/src/WandleWheelhouse.Api/DTOs/Blog/BlogArticleResponseDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System;

namespace WandleWheelhouse.Api.DTOs.Blog;

#nullable enable
public class BlogArticleResponseDto
{
    public Guid BlogArticleId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty; // Include full content
    public string? Excerpt { get; set; }
    public string? Caption { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime PublicationDate { get; set; }
    public DateTime LastUpdatedDate { get; set; }
    public bool IsPublished { get; set; }
    public string? Slug { get; set; }

    // Author details
    public string AuthorId { get; set; } = string.Empty;
    public string? AuthorFullName { get; set; } // e.g., "John Doe"
}
#nullable disable

File: WandleWheelhouse/client/src/dto/Blog/BlogArticleResponseDto.ts
// src/dto/Blog/BlogArticleResponseDto.ts

// Represents the full details of a Blog Article from the API
export interface BlogArticleResponseDto {
    blogArticleId: string; // Assuming GUID becomes string
    title: string;
    content: string; // Full content
    excerpt?: string | null;
    caption?: string | null;
    imageUrl?: string | null;
    publicationDate: string; // ISO Date string
    lastUpdatedDate: string; // ISO Date string
    isPublished: boolean;
    slug?: string | null;
  
    // Author details (matching backend DTO)
    authorId: string;
    authorFullName?: string | null;
    authorAvatarUrl?: string | null; // <-- Add this
  }